thetis/solver2d.py

```diff
@@ -509,14 +509,14 @@ def get_tracer_timestepper(self, integrator, label):
             'tracer_advective_velocity_factor': self.options.tracer_advective_velocity_factor,
         }
 
-        args = (self.equations[label], self.fields[label], fields, self.dt, )
+        args = (self.equations[label], self.fields[label], fields, self.dt)
         kwargs = dict(solver_parameters=self.options.timestepper_options.solver_parameters_tracer)
         if label in self.bnd_functions:
-            kwargs['bnd_conditions'] = self.bnd_functions[label]
+            args += (self.bnd_functions[label],)
         elif label[:-3] in self.bnd_functions:
-            kwargs['bnd_conditions'] = self.bnd_functions[label[:-3]]
+            args += (self.bnd_functions[label[:-3]],)
         else:
-            kwargs['bnd_conditions'] = {}
+            args += ({},)
```

Comment:
maybe cleaner to avoid appending to `args`, so something like:
```python
bcs = {}
if label in self.bnd_functions:
    bcs = self.bnd_functions[label]
args = (self.equations[label], self.fields[label], fields, self.dt, bcs)
```